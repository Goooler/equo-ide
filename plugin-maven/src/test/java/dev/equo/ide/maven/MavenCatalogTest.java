/*******************************************************************************
 * Copyright (c) 2022-2023 EquoTech, Inc. and others.
 *
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     EquoTech, Inc. - initial API and implementation
 *******************************************************************************/
package dev.equo.ide.maven;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import java.io.IOException;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

@ExtendWith({SnapshotExtension.class})
public class MavenCatalogTest extends MavenHarness {
	private void test(String content, Expect expect) throws IOException, InterruptedException {
		setPom(
				content
						+ "\n"
						+ "<filters>\n"
						+ "  <filter><platformNone>true</platformNone></filter>\n"
						+ "</filters>\n");
		mvnw("equo-ide:list -Drequest")
				.snapshotBetween(
						"(default-cli) @ equo-maven-test-harness ---", "[INFO] BUILD SUCCESS", expect);
	}

	@Test
	public void simple(Expect expect) throws Exception {
		test("<platform><version>4.27</version></platform><jdt/>", expect.scenario("jdt"));
		test(
				"<platform><version>4.27</version></platform><gradleBuildship/>",
				expect.scenario("gradleBuildship"));
	}

	@Test
	public void versionOverride(Expect expect) throws IOException, InterruptedException {
		test("<jdt><version>4.25</version></jdt>", expect.scenario("jdt-spec"));
		test(
				"<platform/><jdt><version>4.25</version></jdt>",
				expect.scenario("platform-neutral-jdt-spec"));
		test("<platform><version>4.25</version></platform><jdt/>", expect.scenario("platform-spec"));
		test(
				"<platform><version>4.25</version></platform><jdt><version>4.25</version></jdt>",
				expect.scenario("both-spec"));
	}

	@Test
	public void urlOverride(Expect expect) throws IOException, InterruptedException {
		test("<jdt><url>http://test.url/</url></jdt>", expect);
	}
}
