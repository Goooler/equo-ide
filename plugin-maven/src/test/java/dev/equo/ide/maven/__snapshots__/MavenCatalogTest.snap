dev.equo.ide.maven.MavenCatalogTest.simple[gradleBuildship]=[
+-------------+----------------------------------------------------------------------------------------+
| kind        | value                                                                                  |
+-------------+----------------------------------------------------------------------------------------+
| p2repo      | https://download.eclipse.org/buildship/updates/e423/releases/3.x/3.1.6.v20220511-1359/ |
| p2repo      | https://download.eclipse.org/eclipse/updates/4.26/                                     |
| install     | org.eclipse.buildship.feature.group                                                    |
| install     | org.eclipse.platform.ide.categoryIU                                                    |
| install     | org.eclipse.releng.java.languages.categoryIU                                           |
| filter      | maven-filter-unnamed--1003173200                                                       |
|   osgi.arch | dont-include-platform-specific-artifacts                                               |
|   osgi.os   | dont-include-platform-specific-artifacts                                               |
|   osgi.ws   | dont-include-platform-specific-artifacts                                               |
+-------------+----------------------------------------------------------------------------------------+

[INFO] ------------------------------------------------------------------------
]


dev.equo.ide.maven.MavenCatalogTest.simple[jdt]=[
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.26/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | maven-filter-unnamed--1003173200                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+

[INFO] ------------------------------------------------------------------------
]